paper-api/src/main/java/org/bukkit/inventory/ItemStack.java

```diff
@@ -1306,6 +1306,29 @@ public void resetData(final io.papermc.paper.datacomponent.@NotNull DataComponen
         this.craftDelegate.resetData(type);
     }
 
+    /**
+     * Copies the values of components from the provided ItemStack. Will only copy the value of components from the
+     * source ItemStack that are set or overridden.
+     *
+     * @param source the item stack to copy from
+     */
+    @org.jetbrains.annotations.ApiStatus.Experimental
+    public void copyDataFrom(final @NotNull ItemStack source) {
+        this.copyDataFrom(source, null);
+    }
+
+    /**
+     * Copies the values of components from the provided ItemStack. Will only copy the value of components from the
+     * source ItemStack that are set or overridden.
+     *
+     * @param source the item stack to copy from
```

Comment:
Maybe a code example here would be nice, something that shows you can just create a constant list of types, and then use `THAT_LIST::contains` as the parameter in order to nicely copy data from a specific list

Reply:
Added an example.